db/db_test.cc

```diff
@@ -4911,8 +4914,12 @@ TEST_F(DBTest, DynamicLevelCompressionPerLevel) {
   ASSERT_OK(dbfull()->TEST_WaitForCompact());
   ASSERT_EQ(NumTableFilesAtLevel(1), 0);
   ASSERT_EQ(NumTableFilesAtLevel(2), 0);
-  ASSERT_LT(SizeAtLevel(0) + SizeAtLevel(3) + SizeAtLevel(4),
-            120U * 4000U + 50U * 24);
+  // The default compaction method snappy does not work on Alpine 32 platform
```

Comment:
I'd prefer not to add a new requirement that we make our unit tests work without snappy. Explain "does not work"

Reply:
Snappy does not work in Alpine 32 bit platform. That means there is no compression. Here is the data,

DynamicLevelCompressionPerLevel: level0 + level3 + level4 = 500680
db/db_test.cc:4927: Failure
Expected: (allsize) < (120U * 4000U + 50U * 24), actual: 500680 vs 481200
[  FAILED  ] DBTest.DynamicLevelCompressionPerLevel (257 ms)


In Alpine 64 bit platform,

DynamicLevelCompressionPerLevel: level0 + level3 + level4 = 429788  vs 481200
[       OK ] DBTest.DynamicLevelCompressionPerLevel (223 ms)

Reply:
> Snappy does not work in Alpine 32 bit platform

It doesn't compile? It compiles but doesn't compress when enabled? What?

Reply:
It can compile but it doesn't compress the data

Reply:
(Let me know if I miss anything) I hope I am looking at the right web page: https://pkgs.alpinelinux.org/packages?name=snappy&branch=edge. It shows there is a Snappy package for x86 arch. Have you reached out to the package maintainer regarding this @sike-evolve

Reply:
If snappy is not compiled in, we should be exiting above with `if (!Snappy_Supported()) { return; }`. If it is compiled in, we might need to figure out why it's misbehaving.

I suspect it's more likely a test issue where we are ending up with different data on the various levels than expected. You can compare the two resulting DB dirs between the two platforms. (I would put a temporary `assert(false)` here on both platforms to abort the test in the same place without any clean up.)

Reply:
We used alpine 3.8 with Snappy 1.1.4 for testing. It didn't work. I tried the latest Snappy 1.1.9 in Alpine 3.15 today. It works. I will remove the 32 bit checking.

Reply:
We tested Alpine 3.9 works fine. I already removed 32 bit checking. Please review it.